server/aws-lsp-codewhisperer/src/language-server/agenticChat/agenticChatController.ts

```diff
@@ -3048,6 +3070,48 @@ export class AgenticChatController implements ChatHandlers {
         }
     }
 
+    /**
+     * Creates a chat message for modified files data to be sent to the ModifiedFilesTracker
+     */
+    async #getModifiedFilesDataChatResult(
+        toolUse: ToolUse,
+        doc: TextDocument | undefined,
+        session: ChatSessionService
+    ): Promise<ChatMessage> {
+        const input = toolUse.input as unknown as FsWriteParams | FsReplaceParams
+        const oldContent = session.toolUseLookup.get(toolUse.toolUseId!)?.fileChange?.before ?? ''
+        // Get just the filename instead of the full path
+        const fileName = path.basename(input.path)
+        const diffChanges = diffLines(oldContent, doc?.getText() ?? '')
+        const changes = diffChanges.reduce(
+            (acc, { count = 0, added, removed }) => {
+                if (added) {
+                    acc.added += count
+                } else if (removed) {
+                    acc.deleted += count
+                }
```

Comment:
i believe its also possible for a change to be neither added nor removed (i.e. it is context)